diag_manager/fms_diag_reduction_methods.F90

```diff
@@ -0,0 +1,102 @@
+!> \author Ganga Purja Pun
+!> \email GFDL.Climate.Model.Info@noaa.gov
+!! \brief Contains routines for the modern diag manager
+!! These routines are meant to be used for checks and in reduction methods.
+
+module fms_diag_reduction_methods_mod
+  use platform_mod, only: r8_kind, r4_kind
+  implicit none
+  private
+
+  public :: check_indices_order, init_mask, set_weight
+
+  contains
+
+  !> @brief Checks improper combinations of is, ie, js, and je.
+  !! @return The error message, empty string if no errors were found
+  !> @note accept_data works in either one or another of two modes.
+  !! 1. Input field is a window (e.g. FMS physics)
+  !! 2. Input field includes halo data
+  !! It cannot handle a window of data that has halos.
+  !! (A field with no windows or halos can be thought of as a special case of either mode.)
+  !! The logic for indexing is quite different for these two modes, but is not clearly separated.
+  !! If both the beggining and ending indices are present, then field is assumed to have halos.
+  !! If only beggining indices are present, then field is assumed to be a window.
+  !> @par
+  !! There are a number of ways a user could mess up this logic, depending on the combination
+  !! of presence/absence of is,ie,js,je. The checks below should catch improper combinations.
```

Comment:
All of this should be part of a \\description and not a \\note.  Then you can get rid of \\brief.

Reply:
that is not a description of the function though